### PR TITLE
[Merged by Bors] - chore: conjTranspose and transpose are obviously injective 

### PR DESCRIPTION
**Pull request by MohanadAhmed**
`conjTranspose` and `transpose` are injective

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Matrix/Basic.lean [hunk 1]
```diff
@@ -2017,6 +2017,15 @@ theorem transpose_map {f : α → β} {M : Matrix m n α} : Mᵀ.map f = (M.map
   rfl
 #align matrix.transpose_map Matrix.transpose_map
 
+theorem transpose_inj (A B : Matrix m n α) : Aᵀ = Bᵀ → A = B :=
+  (fun h => ext fun i j => by simpa [transpose_apply] using ext_iff.2 h j i)
+
+theorem tranpose_injecitve : Function.Injective (fun M : Matrix m n α => Mᵀ) :=
+  fun _ _ => transpose_inj _ _
+
+theorem transpose_eq_transpose_iff (A B : Matrix m n α) : Aᵀ = Bᵀ ↔ A = B := by
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/Basic.lean:2026]: This one should be called `transpose_inj`

File: Mathlib/Data/Matrix/Basic.lean [hunk 2]
```diff
@@ -2017,6 +2017,15 @@ theorem transpose_map {f : α → β} {M : Matrix m n α} : Mᵀ.map f = (M.map
   rfl
 #align matrix.transpose_map Matrix.transpose_map
 
+theorem transpose_inj (A B : Matrix m n α) : Aᵀ = Bᵀ → A = B :=
+  (fun h => ext fun i j => by simpa [transpose_apply] using ext_iff.2 h j i)
+
+theorem tranpose_injecitve : Function.Injective (fun M : Matrix m n α => Mᵀ) :=
+  fun _ _ => transpose_inj _ _
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/Basic.lean:2024]: Some typos; it's also better to state it without the `fun`:
```suggestion
theorem transpose_injective : Function.Injective (transpose : Matrix m n α → Matrix n m α) :=
  fun _ _ => transpose_inj _ _
```

File: Mathlib/Data/Matrix/Basic.lean [hunk 3]
```diff
@@ -2017,6 +2017,13 @@ theorem transpose_map {f : α → β} {M : Matrix m n α} : Mᵀ.map f = (M.map
   rfl
 #align matrix.transpose_map Matrix.transpose_map
 
+theorem transpose_inj (A B : Matrix m n α) : Aᵀ = Bᵀ ↔ A = B :=
+  ⟨(fun h => ext fun i j => by simpa [transpose_apply] using ext_iff.2 h j i),
+    fun h => congr_arg transpose h ⟩
+
+theorem transpose_injective : Function.Injective (transpose : Matrix m n α → Matrix n m α) :=
```
Comment (eric-wieser) [inline Mathlib/Data/Matrix/Basic.lean:2024]: It will be easier to prove these in the other order using `eq_iff`

Comment (MohanadAhmed) [reply Mathlib/Data/Matrix/Basic.lean:2024]: Which kind of easy are we talking about. They are both one liners, and I according to my understanding lean compiles a theorem it's proof is irrelevant?

**Review by eric-wieser — APPROVED** [review]
I pushed some golfs; it turns out these lemmas would have simplified my previous PR!

bors d+

Thanks!

**Pull request closed without merging**